### PR TITLE
Change minimal Node.js version to 10

### PR DESCRIPTION
**Pull request by puzpuzpuz**
* Change minimal Node.js version to 10
  - Reason: as Node.js 8 is no longer supported and gets no security patches, we shouldn't support it
* Updates dev dependency versions
* Fixes linter errors and disables some warnings
* Also fixes potential unhandled promise rejection in `ReliableTopicProxy#trySendMessage`

I've also changed Node.js version to 10 in the Jenkins CI job

### CODE REVIEW

**Comment by puzpuzpuz** [comment]
verify

File: package.json [hunk 1]
```diff
@@ -11,26 +11,26 @@
...
         "mousse": "^0.3.1",
         "nyc": "^15.1.0",
         "rimraf": "^3.0.2",
-        "sinon": "^9.0.2",
+        "sinon": "^9.0.3",
         "thrift": "^0.12.0",
         "typedoc": "^0.18.0",
-        "typescript": "^3.9.5",
-        "winston": "^3.2.1"
+        "typescript": "^3.9.7",
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply package.json:29]: In future, yes. At the moment `@typescript-eslint/eslint-plugin` doesn't support TS 4.0, so it's better to stick to 3.x for now.

[the remark this answers is not in the document]
Comment (mdumandag) [reply package.json:29]: Oh, okay. Thanks for the explanation.

File: package.json [hunk 2]
```diff
@@ -11,26 +11,26 @@
         "@types/long": "4.0.1"
     },
     "devDependencies": {
-        "@types/node": "^8.10.60",
-        "@typescript-eslint/eslint-plugin": "^2.34.0",
-        "@typescript-eslint/parser": "^2.34.0",
+        "@types/node": "^10.17.28",
+        "@typescript-eslint/eslint-plugin": "^3.9.1",
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply package.json]: Good point. The package was published after this PR was created. Changed versions in 17044796e100153a871f609f537305c49a08d6ee

**Review by mdumandag — APPROVED** [review]

**Pull request merged**